### PR TITLE
2021 Clarity Challenge recipe

### PR DESCRIPTION
**Pull request by popcornell**
This is an attempt to do a simple recipe for 2021 Clarity Challenge. 

Currently missing: test set (I need to check if the oracle sources have been released for test set)
Also where do you set in recipes the path to the current root of the dataset ? What's the best practice ?

### CODE REVIEW

File: egs2/clarity/enh_2021/local/data.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+set -e
+set -u
+set -o pipefail
+
+
+python local/prep_data.py --clarity_root /raid/users/popcornell/Clarity/target_dir/clarity_CEC1_data/clarity_data/
```
Comment (Emrys365) [inline egs2/clarity/enh_2021/local/data.sh:8]: Could you replace this hard-coded data path with a variable?

Comment (Emrys365) [inline egs2/clarity/enh_2021/local/data.sh:8]: Could you replace this hard-coded data path with a variable?

File: egs2/clarity/enh_2021/run.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+sample_rate=44k
+# Path to a directory containing extra annotations for CHiME4
+# Run `local/data.sh` for more information.
+extra_annotations=
```
Comment (Emrys365) [inline egs2/clarity/enh_2021/run.sh:11]: This variables can be removed because they are intended for CHiME4.

File: egs2/clarity/enh_2021/run.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+./enh.sh \
+    --train_set "${train_set}" \
+    --valid_set "${valid_set}" \
+    --test_sets "${test_sets}" \
+    --fs ${sample_rate} \
+    --ngpu 1 \
+    --spk_num 1 \
+    --ref_channel 0 \
+    --local_data_opts "--extra-annotations ${extra_annotations} --stage 1 --stop-stage 2" \
```
Comment (Emrys365) [inline egs2/clarity/enh_2021/run.sh:27]: ditto

Comment (Emrys365) [inline egs2/clarity/enh_2021/run.sh:27]: ditto

File: egs2/clarity/enh_2021/run.sh [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    --enh_config conf/tuning/train_enh_beamformer_mvdr.yaml \
+    --use_dereverb_ref false \
+    --use_noise_ref true \
```
Comment (Emrys365) [inline egs2/clarity/enh_2021/run.sh:30]: Maybe it's better to set this to `false` if no parallel noise data is provided for training.

Comment (Emrys365) [inline egs2/clarity/enh_2021/run.sh:30]: Maybe it's better to set this to `false` if no parallel noise data is provided for training.

File: egs2/clarity/enh_2021/local/data.sh (hunk 1 above)
Comment (Emrys365) [reply egs2/clarity/enh_2021/local/data.sh:8]: It would be great if you could add some information about where to download the data.

Comment (popcornell) [reply egs2/clarity/enh_2021/local/data.sh]: done ! Thanks

File: egs2/clarity/enh_2021/run.sh (hunk 4 above)
Comment (popcornell) [reply egs2/clarity/enh_2021/run.sh:29]: I ve added the interferer ref now in the scp files

**Comment by popcornell** [comment]
I have also results

# RESULTS
## Environments
- date: `Tue Apr  5 16:36:29 UTC 2022`
- python version: `3.9.7 (default, Sep 16 2021, 13:09:58)  [GCC 7.5.0]`
- espnet version: `espnet 0.10.5a1`
- pytorch version: `pytorch 1.10.1`
- Git hash: `46eaa5eb6bea11cc33927392dca7888921491d8c`
  - Commit date: `Sat Mar 26 22:35:44 2022 +0100`


## enh_train_enh_beamformer_mvdr_raw

config: conf/tuning/train_enh_beamformer_mvdr.yaml

|dataset|STOI|SAR|SDR|SIR|
|---|---|---|---|---|
|enhanced_dev|0.94|11.74|11.74|0.00|

**Comment by popcornell** [comment]
Tests fails but probably not related to this PR

File: espnet2/bin/enh_scoring.py [hunk 5]
```diff
@@ -63,18 +63,14 @@ def scoring(
...
                 # multi-channel reference and single-channel output
                 ref = ref[..., ref_channel]
-                assert ref.shape == inf.shape, (ref.shape, inf.shape)
             elif ref.ndim < inf.ndim:
                 # single-channel reference and multi-channel output
-                raise ValueError(
-                    "Reference must be multi-channel when the \
-                    network output is multi-channel."
-                )
+                inf = inf[..., ref_channel]
```
Comment (popcornell) [inline espnet2/bin/enh_scoring.py:68]: @Emrys365 I changed this because on evaluation only anechoic signals from channel 0 are provided in this dataset.

Comment (Emrys365) [reply espnet2/bin/enh_scoring.py:68]: Oh I see. Indeed, this case should be supported in evaluation.

File: egs2/clarity/enh_2021/local/data.sh [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    [--sample_rate]: 16000 (default) or 44100
+EOF
+)
+
+clarity_root=
+sample_rate=16000
+
+. utils/parse_options.sh
+
+
```
Comment (Emrys365) [inline egs2/clarity/enh_2021/local/data.sh:25]: Maybe add a check for the existence of `sox` (as it is assumed to be available in `local/prep_data.py`) like the following?
```bash
! command -v sox &>/dev/null && echo "sox: command not found" && exit 1;
```

File: egs2/clarity/enh_2021/local/prep_data.py [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+
+parser = argparse.ArgumentParser("Clarity")
+parser.add_argument(
+    "--clarity_root",
+    type=str,
+    help="Path to Clarity Challenge root folder "
+    "(Folder containing train, dev and metadata dirs)",
+)
+parser.add_argument(
+    "--fs",
```
Comment (Emrys365) [inline egs2/clarity/enh_2021/local/prep_data.py:14]: Is this argument used?

File: egs2/clarity/enh_2021/local/prep_data.py [hunk 8]
```diff
@@ -0,0 +1,130 @@
+import argparse
+import os
+import json
+from pathlib import Path
```
Comment (Emrys365) [inline egs2/clarity/enh_2021/local/prep_data.py:4]: Could you fix the import order to follow the alphabet order?

**Comment by Emrys365** [comment]
BTW, You may need to merge the latest master branch to fix the CI test error.

File: egs2/clarity/enh_2021/local/prep_data.py (hunk 7 above)
Comment (popcornell) [reply egs2/clarity/enh_2021/local/prep_data.py:14]: yes it is used now

File: egs2/clarity/enh_2021/local/prep_data.py (hunk 8 above)
Comment (popcornell) [reply egs2/clarity/enh_2021/local/prep_data.py:4]: done

**Comment by popcornell** [comment]
Thanks for your suggestions ! Now it should be better.

**Review by Emrys365 — APPROVED** [review]
LGTM now. Thanks!

**Comment by sw005320** [comment]
Can you upload a model to the HF hub and add a link to README.md?

**Comment by popcornell** [comment]
Sure, I'll look into how to do it

**Comment by popcornell** [comment]
> Could you rename the recipe to clarity21 to distinguish from the challenges in other years?

Isn't it better to use a subfolder as I am using right now ? I am using `enh_2021` subfolder

**Comment by Emrys365** [comment]
> > Could you rename the recipe to clarity21 to distinguish from the challenges in other years?
> 
> Isn't it better to use a subfolder as I am using right now ? I am using `enh_2021` subfolder

It seems that espnet prefers to split them into different recipes, like chime4 and chime5. So it's better to also follow the convention here.

File: egs2/clarity21/enh_2021/README.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+## Environments
+- date: `Tue Apr 12 20:54:54 UTC 2022`
+- python version: `3.9.7 (default, Sep 16 2021, 13:09:58)  [GCC 7.5.0]`
+- espnet version: `espnet 0.10.5a1`
+- pytorch version: `pytorch 1.10.1`
+- Git hash: `46eaa5eb6bea11cc33927392dca7888921491d8c`
+  - Commit date: `Sat Mar 26 22:35:44 2022 +0100`
+
+
+## ..
```
Comment (sw005320) [inline egs2/clarity21/enh_2021/README.md:21]:
```suggestion
## enh_train_enh_beamformer_mvdr_raw
```

**Comment by sw005320** [comment]
Thanks, @popcornell!
